### DISCUSSION TITLE
Functionality Questions

### DISCUSSION THREAD

**Discussion by ICanDoThings2** [JSBSim-Team/jsbsim#621] [Q&A]
I had some questions about what JSBSim can/can't do. I noticed there was a PointMass class, http://jsbsim.sourceforge.net/JSBSim/classJSBSim_1_1FGMassBalance.html#details detailed some here. Can this be used to change the mass of the aircraft at runtime? Like, making it drop or take off with things to change it's base mass?

**Comment by seanmcleod** [comment]
Sure you can change the mass of a point-mass at runtime, e.g. set it' mass to 0 to simulate it being dropped, I don't see an explicit delete mechanism.

https://github.com/JSBSim-Team/jsbsim/blob/59c0bd909dbe7232e77ac9dc03eb65f959f16cf9/src/models/FGMassBalance.cpp#L431-L448

**Reply by ICanDoThings2** [reply]
Ah! That might work. I wondered if it was as simple as removing it from the vector or inserting it, however. It does say here http://jsbsim.sourceforge.net/JSBSim/classJSBSim_1_1FGMassBalance.html#details it maintains a vector of all the pointmass, so I might think? I suppose it would require testing, wouldn't it?

**Comment by seanmcleod** [comment]
My guess is it is as easy as removing an entry from the vector, although given the property bindings that are created for each instance you may need to remove the property bindings as well. However, the vector is a private member variable and there is no public accessor.

So you have a couple of choices. 

Just set the mass of the point mass to 0.

In your own fork create a public accessor, or a specific `DeletePointMass()` method and maintain your fork merging in any new JSBSim updates going forward that you might want.

Submit a pull request that adds a `DeletePointMass()` method which if it's accepted means your code going forward can rely on it without you having to maintain your own fork.

**Comment by jonsberndt** [comment]
As Sean points out (in the link he refers to), you can change the location and weight of a pointmass at runtime. I have used this in the past to simulate a person getting into and out of an aircraft sitting on the runway – there was a specific reason for that test. You could also use it to model dropping  a load of food and medical supplies (😊) or whatever while in-flight.

Looking at the code (and in retrospect) changing the *weight* of a pointmass might have been a bad design choice, as opposed to changing the *mass* of a pointmass.

 

Jon

**Comment by ICanDoThings2** [comment]
Ah. That is a shortcoming I was concerned about. It's nice to know my issue wasn't unfounded, I think I will try modding JSBSim further to my needs. I'm grateful to have a helpful and supportive community like you two though, thanks!

**Comment by ICanDoThings2** [comment]
I was thinking since I guarantee this won't be my last discussion here about JSBSim; would anyone else be interested in a Discord about JSBSim?

**Reply by ICanDoThings2** [reply]
https://discord.gg/799myYBn I made a server here.
